receiver/elasticapmreceiver/internal/mappers/intakeV2ToDerivedFields.go

```diff
@@ -89,3 +90,69 @@ func SetDerivedFieldsCommon(event *modelpb.APMEvent, attributes pcommon.Map) {
 		attributes.PutStr(elasticattr.EventOutcome, "unknown")
 	}
 }
+
+// Sets fields that are NOT part of OTel for errors. These fields are derived by the Enrichment lib in case of OTLP input
+func SetDerivedFieldsForError(event *modelpb.APMEvent, attributes pcommon.Map) {
```

Comment:
nit: an anonymous function could be used to wrap the string to nil comparison and attributes setter, similar to https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/geoipprocessor/internal/provider/maxmindprovider/provider.go#L71